### PR TITLE
Use mCurrentFocus along with mFocusedApp to detect the focused activity

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/4489

### CODE REVIEW

File: lib/tools/apk-utils.js [hunk 1]
```diff
@@ -141,26 +141,47 @@ apkUtilsMethods.startApp = async function (startAppOptions = {}) {
...
-  let searchRe = new RegExp('mFocusedApp.+Record\\{.*\\s([^\\s\\/\\}]+)' +
-                            '\\/([^\\s\\/\\}\\,]+)\\,?(\\s[^\\s\\/\\}]+)*\\}'); // https://regex101.com/r/xZ8vF7/1
+  const cmd = ['dumpsys', 'window', 'windows'];
+  const nullFocusedAppRe = new RegExp(/mFocusedApp=null/);
+  // https://regex101.com/r/xZ8vF7/1
+  const focusedAppRe = new RegExp('mFocusedApp.+Record\\{.*\\s([^\\s\\/\\}]+)' +
+                                        '\\/([^\\s\\/\\}\\,]+)\\,?(\\s[^\\s\\/\\}]+)*\\}');
+  const nullCurrentFocusRe = new RegExp(/mCurrentFocus=null/);
+  const currentFocusAppRe = new RegExp('mCurrentFocus.+\\{\\w+\\s+([^\\s\\/]+)\\/([^\\s]+)');
+  const getMatch = (pattern, output) => {
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: Does this have advantages over a single multiline regular expression?

Comment (mykola-mokhnach) [reply lib/tools/apk-utils.js]: better readability and easier to change IMHO

Comment (imurchie) [reply lib/tools/apk-utils.js]: More readable than `m`? I'm not suggesting combining the four regular expressions. I'm just wondering why split the output and run the regular expression a bunch of times. Each regular expression can be run a single time against the entire output.

Comment (jlipps) [reply lib/tools/apk-utils.js]: yeah, why not just get the first match in a multiline regex test?

Comment (mykola-mokhnach) [reply lib/tools/apk-utils.js]: got your point. Changed

**Comment by mykola-mokhnach** [comment]
Is it ok now @imurchie ?

**Pull request merged**